src/ApiResource/Error.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\ApiResource;
+
+use ApiPlatform\Exception\ErrorExceptionInterface;
+use ApiPlatform\Metadata\ApiProperty;
+use ApiPlatform\Metadata\NotExposed;
+
+#[NotExposed(uriTemplate: '/errors/{statusCode}')]
+class Error
+{
+    private string $title;
+
+    private string $description;
+
+    #[ApiProperty(identifier: true)]
+    private int $statusCode;
+
+    private array $trace;
+
+    public function __construct(ErrorExceptionInterface $exception)
```

Comment:
Use public readonly promoted constructor parameters instead of private with getter properties.